src/FondOfSpryker/Yves/GoogleTagManager/Plugin/VariableBuilder/ProductVariables/ProductFieldPriceExcludingTaxPlugin.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace FondOfSpryker\Yves\GoogleTagManager\Plugin\VariableBuilder\ProductVariables;
+
+use Exception;
+use FondOfSpryker\Yves\GoogleTagManager\Dependency\VariableBuilder\ProductFieldVariableBuilderPluginInterface;
+use Generated\Shared\Transfer\GooleTagManagerProductDetailTransfer;
+use Generated\Shared\Transfer\ProductAbstractTransfer;
+use Spryker\Shared\Log\LoggerTrait;
+use Spryker\Yves\Kernel\AbstractPlugin;
+
+/**
+ * @method \FondOfSpryker\Yves\GoogleTagManager\GoogleTagManagerFactory getFactory()
+ */
+class ProductFieldPriceExcludingTaxPlugin extends AbstractPlugin implements ProductFieldVariableBuilderPluginInterface
+{
+    use LoggerTrait;
+
+    /**
+     * @param \Generated\Shared\Transfer\GooleTagManagerProductDetailTransfer $gooleTagManagerProductDetailTransfer
+     * @param \Generated\Shared\Transfer\ProductAbstractTransfer $product
+     * @param array $params
+     *
+     * @return \Generated\Shared\Transfer\GooleTagManagerProductDetailTransfer
+     */
+    public function handle(
+        GooleTagManagerProductDetailTransfer $gooleTagManagerProductDetailTransfer,
+        ProductAbstractTransfer $product,
+        array $params = []
+    ): GooleTagManagerProductDetailTransfer {
+        try {
```

Comment:
Machts nicht eher Sinn den Try Catch in die Caller Class zu packen?

Reply:
Weiß ich nicht, dass betrifft dann alle Plugins. @sourcecube was meinst du?

Reply:
Hmm ja vielleicht ist es sinniger das überlegt abzufangen, so das jedes plugin geschützt ist. Ansonsten baut einer ohne schutz und alles fliegt weg.